synapseclient/__main__.py

```diff
@@ -977,54 +977,66 @@ def login_with_prompt(syn, user, password, rememberMe=False, silent=False, force
         _authenticate_login(syn, user, password, silent=silent, rememberMe=rememberMe, forced=forced)
     except SynapseNoCredentialsError:
         # if there were no credentials in the cache nor provided, prompt the user and try again
-        while not user:
-            user = input("Synapse username: ")
```

Comment:
I've updated the prompt slightly, if they supply a username then we don't prompt for one but the username appears as part of the password prompt (i.e. "Password, api key, or auth token for user {user}:"), and indicated that if they do happen to input a username they can still supply an auth token if they choose (although it will error if it doesn't match the username).